src/Stoa.ts

```diff
@@ -159,6 +161,42 @@ class Stoa {
                 res.status(500).send("An error occurred while saving");
             });
         });
+
+        /**
+         * When a request is received through the `/preimage_received` handler
+         * JSON preImage data is parsed and stored on each storage.
+         */
+        this.stoa.post("/preimage_received",
+            (req: express.Request, res: express.Response, next: express.NextFunction) => {
+
+            let pre_image: PreImageInfo;
+            if (req.body.pre_image == undefined)
+            {
+                res.status(400).send("Missing 'preImage' object in body");
+            }
+
+            try {
+                if (typeof req.body.pre_image === "string")
+                    pre_image = JSON.parse(req.body.pre_image);
+                else
+                    pre_image = req.body.pre_image;
```

Comment:
Why is this needed ?

Reply:
It was maintained for compatibility with the previous test, Agora History Push.
But now It doesn't need it
I will change it.
`pre_image = req.body.pre_image;`